man/systemd.exec.xml

```diff
@@ -381,6 +381,14 @@
         definitions. Each definition consists of a colon-separated tuple of source path and destination
         definitions, optionally followed by another colon and a list of mount options.</para>
 
+        <para>If the destination is empty, then a variable overlay is set up on the root. Nodes in the image
+        that are leaves in the root (empty directories or files) are bind-mounted individually.
+        Existing directories with differing content will be set up with a read-only overlayfs.
+        For example, if the root has <filename>/opt/</filename> and <filename>/var/foo</filename>, and the
+        image has <filename>/opt/bar</filename> and <filename>/var/baz</filename>, the end result will be
+        that <filename>/opt/</filename> from the image is bind-mounted on <filename>/opt/</filename> on the root,
+        and <filename>/var</filename> from the root and the image will both be part of an OverlayFS mount.</para>
+
```

Comment:
hmm, it's not clear to me, why the destination being unspecified should mean "overlayfs"? i.e. why wouldn't i want to use overlayfs logic with a specific specified destination path?

i.e. shouldn't this be a new option OverlayMount= or so?

Reply:
The overlay logic to me makes sense root-on-root, so no need for a dest, and for MountImages empty dest doesn't make sense. So they fit together and complement, so to speak. Implementation is a bit easier too, I think. Do you see cases where it would be useful on a subtree?

Reply:
Moved to new directive RootExtensions as discussed